### ISSUE TITLE
[Bug] Crash after room destruction

### ISSUE THREAD

**Issue by darcyrush** [CorsixTH/CorsixTH#1957]
#### Describe the issue
Earthquakes caused equipment to explode while doctors and patients were using them. Removed room and tried to rebuild, froze soon after

#### Save Game
[Autosave.zip](https://github.com/CorsixTH/CorsixTH/files/6704134/Autosave.zip)
Either 9 or 10

#### System Information
**CorsixTH Version:** *22fcffddb2*

**Operating System:** *Ubuntu 20.04*

**Theme Hospital Version:** *GOG.com*

#### Gamelog.txt
```
Created game with savegame version 156.
Warning: Trying to remove nonexistent room build callback (function: 0x56235376c490) from humanoid (table: 0x5623537bf410).
Warning: Trying to remove nonexistent room remove callback (function: 0x56235272d7b0) from humanoid (table: 0x5623537bf410).
Warning: Trying to remove nonexistent staff callback (function: 0x5623539cf9c0) from humanoid (table: 0x5623537bf410).
Warning: Trying to remove nonexistent room build callback (function: 0x5623547ed4c0) from humanoid (table: 0x5623543ad450).
Warning: Trying to remove nonexistent room remove callback (function: 0x5623527d5270) from humanoid (table: 0x5623543ad450).
Warning: Trying to remove nonexistent staff callback (function: 0x5623542bde60) from humanoid (table: 0x5623543ad450).
Warning: Trying to remove nonexistent room build callback (function: 0x562354f53190) from humanoid (table: 0x562352b386d0).
Warning: Trying to remove nonexistent room remove callback (function: 0x5623538824b0) from humanoid (table: 0x562352b386d0).
Warning: Trying to remove nonexistent staff callback (function: 0x562352f4e720) from humanoid (table: 0x562352b386d0).
Warning: Trying to remove nonexistent room build callback (function: 0x562353649f30) from humanoid (table: 0x562354e65dc0).
Warning: Trying to remove nonexistent room remove callback (function: 0x562352b12a00) from humanoid (table: 0x562354e65dc0).
Warning: Trying to remove nonexistent staff callback (function: 0x5623550e9d00) from humanoid (table: 0x562354e65dc0).
Warning: Trying to remove nonexistent room build callback (function: 0x562353588eb0) from humanoid (table: 0x562353e5a3b0).
Warning: Trying to remove nonexistent room remove callback (function: 0x562353531fb0) from humanoid (table: 0x562353e5a3b0).
Warning: Trying to remove nonexistent staff callback (function: 0x562352df4af0) from humanoid (table: 0x562353e5a3b0).
Warning: Trying to remove nonexistent room build callback (function: 0x562354869320) from humanoid (table: 0x562353174570).
Warning: Trying to remove nonexistent room remove callback (function: 0x5623538fce70) from humanoid (table: 0x562353174570).
Warning: Trying to remove nonexistent staff callback (function: 0x562353522980) from humanoid (table: 0x562353174570).
Savegame version is 156 (v0.65), originally it was 156 (v0.65).
Warning: Trying to remove nonexistent room build callback (function: 0x55cb8057e230) from humanoid (table: 0x55cb81341e60).
Warning: Trying to remove nonexistent room remove callback (function: 0x55cb8169f770) from humanoid (table: 0x55cb81341e60).
Warning: Trying to remove nonexistent staff callback (function: 0x55cb8080c1e0) from humanoid (table: 0x55cb81341e60).
Error in timer handler: 
...ines/CorsixTH/CorsixTH/Lua/hospitals/player_hospital.lua:344: attempt to get length of a nil value (field 'queue')
stack traceback:
        ...ines/CorsixTH/CorsixTH/Lua/hospitals/player_hospital.lua:344: in method 'warnForLongQueues'
        ...ines/CorsixTH/CorsixTH/Lua/hospitals/player_hospital.lua:184: in method 'dailyAdviceChecks'
        ...ines/CorsixTH/CorsixTH/Lua/hospitals/player_hospital.lua:374: in method 'onEndDay'
        .../Workspace/Other/engines/CorsixTH/CorsixTH/Lua/world.lua:1043: in method 'onTick'
        ...user/Workspace/Other/engines/CorsixTH/CorsixTH/Lua/app.lua:1134: in function <...user/Workspace/Other/engines/CorsixTH/CorsixTH/Lua/app.lua:1131>
        (...tail calls...)
        ...user/Workspace/Other/engines/CorsixTH/CorsixTH/Lua/app.lua:1034: in function <...user/Workspace/Other/engines/CorsixTH/CorsixTH/Lua/app.lua:1029>

```

### Additional Info
```
stack traceback:
	...ines/CorsixTH/CorsixTH/Lua/hospitals/player_hospital.lua:344: in method 'warnForLongQueues'
	...ines/CorsixTH/CorsixTH/Lua/hospitals/player_hospital.lua:184: in method 'dailyAdviceChecks'
	...ines/CorsixTH/CorsixTH/Lua/hospitals/player_hospital.lua:374: in method 'onEndDay'
	.../Workspace/Other/engines/CorsixTH/CorsixTH/Lua/world.lua:1043: in method 'onTick'
	...user/Workspace/Other/engines/CorsixTH/CorsixTH/Lua/app.lua:1134: in function <...user/Workspace/Other/engines/CorsixTH/CorsixTH/Lua/app.lua:1131>
	(...tail calls...)
	...user/Workspace/Other/engines/CorsixTH/CorsixTH/Lua/app.lua:1034: in function <...user/Workspace/Other/engines/CorsixTH/CorsixTH/Lua/app.lua:1029>

Warning: No event handler for timer
```

**Comment by lewri** [comment]
Can you try with 0.65... I believe this bug was caught and fixed.

**Comment by lewri** [comment]
Notably should have been found in #1927 , fixed in #1928
